### ISSUE TITLE
`Kokkos::DualView` is still on `<DataType, class, class, class>` instead of `<DataType, Properties...>`

### ISSUE THREAD

**Issue by romintomasetti** [kokkos/kokkos#6085]
**Describe the bug**

It seems that `Kokkos::DualView` is still templated on
```c++
<class DataType, class Arg1Type = void, class Arg2Type = void, class Arg3Type = void>
```
whereas other containers like `OffsetView` are already following the design of `Kokkos::View` with:
```c++
template <class DataType, class... Properties>
```

Here are the changes needed to make `DualView` more consistent if you think it's the right thing to do.
I can do a PR if need be.

```patch
@@ -86,22 +86,21 @@ inline const Kokkos::Cuda& get_cuda_space(const NonCudaExecSpace&) {
 #endif  // KOKKOS_ENABLE_CUDA
 
 }  // namespace Impl
-template <class DataType, class Arg1Type = void, class Arg2Type = void,
-          class Arg3Type = void>
-class DualView : public ViewTraits<DataType, Arg1Type, Arg2Type, Arg3Type> {
-  template <class, class, class, class>
+template <class DataType, class... Properties>
+class DualView : public ViewTraits<DataType, Properties...> {
+  template <class, class...>
   friend class DualView;
 
  public:
   //! \name Typedefs for device types and various Kokkos::View specializations.
   //@{
-  using traits = ViewTraits<DataType, Arg1Type, Arg2Type, Arg3Type>;
+  using traits = ViewTraits<DataType, Properties...>;
 
   //! The Kokkos Host Device type;
   using host_mirror_space = typename traits::host_mirror_space;
 
   //! The type of a Kokkos::View on the device.
-  using t_dev = View<typename traits::data_type, Arg1Type, Arg2Type, Arg3Type>;
+  using t_dev = View<typename traits::data_type, traits>;
 
   /// \typedef t_host
   /// \brief The type of a Kokkos::View host mirror of \c t_dev.
@@ -110,7 +109,7 @@ class DualView : public ViewTraits<DataType, Arg1Type, Arg2Type, Arg3Type> {
   //! The type of a const View on the device.
   //! The type of a Kokkos::View on the device.
   using t_dev_const =
-      View<typename traits::const_data_type, Arg1Type, Arg2Type, Arg3Type>;
+      View<typename traits::const_data_type, traits>;
 
   /// \typedef t_host_const
   /// \brief The type of a const View host mirror of \c t_dev_const.
```

**Comment by ajpowelsnl** [comment]
Many thanks for bringing this to our attention, Romin.  If it's not too much trouble, would you please develop a PR for this issue?

**Comment by ajpowelsnl** [comment]
Hi @romintomasetti and @dalg24 - have all of the necessary changes been merged?  If so, can we close?  If not, can we define steps that still need to be done to complete?

**Comment by dalg24** [comment]
Fixed in #6120

**Comment by masterleinad** [comment]
Fixed in #6120.
